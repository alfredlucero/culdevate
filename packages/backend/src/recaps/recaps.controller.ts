import { Response } from "express";
import { RequestWithUser } from "../interfaces/requestWithUser";
import RecapsDao from "./recaps.dao";
import { Recap } from "./recaps.model";

const RecapsController = {
  async createRecap(req: RequestWithUser, res: Response) {
    const currentUser = req.user;
    const recap: Recap = {
      ...req.body,
      userId: currentUser.id,
    };

    try {
      const createdRecap = await RecapsDao.createRecap(recap);

      res.status(201).json(createdRecap);
    } catch (err) {
      res.status(500).json({ message: "Failed to create recap." });
    }
  },

  async updateRecap(req: RequestWithUser, res: Response) {
    const currentUser = req.user;
    const recapId = req.params.recapId;
    const updatedRecap: Recap = {
      ...req.body,
      userId: currentUser.id,
    };

    try {
      const foundRecap = await RecapsDao.findRecapById(recapId);
      const isMatchingUserId = foundRecap && foundRecap.userId.toString() === currentUser.id;

      if (!foundRecap || !isMatchingUserId) {
        return res.status(404).json({ message: "Failed to find matching recap to update" });
      }

      const changedRecap = await RecapsDao.updateRecapById({ recapId, updatedRecap });

      if (changedRecap) {
        res.status(200).json(changedRecap);
      } else {
        // If by some chance it got deleted while we were trying to update it
        // we'll still get a 404 back
        res.status(404).json({ message: "Failed to find matching recap to update" });
      }
    } catch (err) {
      res.status(500).json({ message: "Failed to update recap" });
    }
  },

  async deleteRecap(req: RequestWithUser, res: Response) {
    const recapId = req.params.recapId;
    const currentUser = req.user;

    try {
      const foundRecap = await RecapsDao.findRecapById(recapId);
      const isMatchingUserId = foundRecap && foundRecap.userId.toString() === currentUser.id;

      if (!foundRecap || !isMatchingUserId) {
        return res.status(404).json({ message: "Failed to find matching recap to delete" });
      }

      const deletedRecap = await RecapsDao.removeRecapById(recapId);

      if (deletedRecap) {
        res.status(200).json(deletedRecap);
      } else {
        // If by some chance it got deleted while we were trying to delete it
        // we'll still get a 404 back
        res.status(404).json({ message: "Failed to find matching recap to delete" });
      }
    } catch (err) {
      res.status(500).json({ message: "Failed to delete recap" });
    }
  },
};

export default RecapsController;
