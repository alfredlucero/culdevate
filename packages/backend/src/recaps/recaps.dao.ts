import {
  RecapBaseModel,
  RecapWorkExperienceModel,
  RecapEducationModel,
  RecapAccomplishmentsModel,
  RecapPublicationsModel,
  RecapSkillsModel,
  RecapSideProjectsModel,
  RecapOrganizationsModel,
  RecapReferencesModel,
  RecapOtherModel,
  Recap,
} from "./recaps.model";

const RecapsDao = {
  findAllRecaps(userId: string) {
    return RecapBaseModel.find({ userId });
  },

  findRecapById(recapId: string) {
    return RecapBaseModel.findById(recapId);
  },

  findRecapByIdAndUserId({ recapId, userId }: { recapId: string; userId: string }) {
    return RecapBaseModel.findOne({ userId, _id: recapId });
  },

  createRecap(recap: Recap) {
    switch (recap.kind) {
      case "Work Experience":
        const workExperienceRecap = new RecapWorkExperienceModel(recap);
        return workExperienceRecap.save();
      case "Education":
        const educationRecap = new RecapEducationModel(recap);
        return educationRecap.save();
      case "Accomplishments":
        const accomplishmentsRecap = new RecapAccomplishmentsModel(recap);
        return accomplishmentsRecap.save();
      case "Publications":
        const publicationsRecap = new RecapPublicationsModel(recap);
        return publicationsRecap.save();
      case "Skills":
        const skillsRecap = new RecapSkillsModel(recap);
        return skillsRecap.save();
      case "Side Projects":
        const sideProjectsRecap = new RecapSideProjectsModel(recap);
        return sideProjectsRecap.save();
      case "Organizations":
        const organizationsRecap = new RecapOrganizationsModel(recap);
        return organizationsRecap.save();
      case "References":
        const referencesRecap = new RecapReferencesModel(recap);
        return referencesRecap.save();
      case "Other":
        const otherRecap = new RecapOtherModel(recap);
        return otherRecap.save();
      default:
        throw new Error("Failed to find matching recap kind to create!");
    }
  },

  updateRecapById({ recapId, updatedRecap }: { recapId: string; updatedRecap: Recap }) {
    switch (updatedRecap.kind) {
      case "Work Experience":
        return RecapWorkExperienceModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "Education":
        return RecapEducationModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "Accomplishments":
        return RecapAccomplishmentsModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Publications":
        return RecapPublicationsModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Skills":
        return RecapSkillsModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Side Projects":
        return RecapSideProjectsModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "Organizations":
        return RecapOrganizationsModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "References":
        return RecapReferencesModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Other":
        return RecapOtherModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      default:
        throw new Error("Failed to find matching recap kind to update!");
    }
  },

  removeRecapById(recapId: string) {
    return RecapBaseModel.findByIdAndDelete({ _id: recapId });
  },
};

export default RecapsDao;
