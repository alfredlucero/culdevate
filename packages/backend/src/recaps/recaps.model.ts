import mongoose from "mongoose";
import { MAX_BULLETPOINT_LENGTH, MAX_BULLETPOINTS, MAX_GENERAL_LENGTH } from "./recaps.validation";

export interface RecapBase {
  kind: RecapKind;
  startDate?: Date;
  endDate?: Date;
  bulletPoints: string[];
}

export type RecapKind =
  | "WorkExperience"
  | "Education"
  | "Accomplishments"
  | "Skills"
  | "SideProjects"
  | "Organizations"
  | "References"
  | "Other";

const recapBaseOptions = {
  discriminatorKey: "kind",
  collection: "recaps",
};

const recapBaseSchema = new mongoose.Schema(
  {
    startDate: { type: Date },
    endDate: { type: Date },
    bulletPoints: {
      type: [
        {
          type: String,
          maxlength: MAX_BULLETPOINT_LENGTH,
        },
      ],
      required: true,
      maxlength: MAX_BULLETPOINTS,
    },
  },
  recapBaseOptions,
);

export interface RecapBaseModel extends RecapBase, mongoose.Document {}

// Base Recap model from which other Recap models will inherit from based on the
// the value of the "kind" discriminator
export const RecapBaseModel = mongoose.model<RecapBaseModel>("Recap", recapBaseSchema);

// Work Experience

export interface RecapWorkExperience {
  title: string;
  location: string;
  company: string;
  employmentType: EmploymentType;
}

export type EmploymentType = "partTime" | "fullTime";

const recapWorkExperienceSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  location: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  company: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  employmentType: { type: String, required: true, enum: ["partTime", "fullTime"] },
});

export interface RecapWorkExperienceModel extends RecapBaseModel, RecapWorkExperience {}

const RecapWorkExperience = RecapBaseModel.discriminator("WorkExperience", recapWorkExperienceSchema);

export const RecapWorkExperienceModel = mongoose.model<RecapWorkExperienceModel>("WorkExperience");

// Education

export interface RecapEducation {
  school: string;
  location: string;
  degree: string;
  fieldOfStudy: string;
  grade: string;
}

const recapEducationSchema = new mongoose.Schema({
  school: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  location: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  degree: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  fieldOfStudy: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  grade: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapEducationModel extends RecapBaseModel, RecapEducation {}

const RecapEducation = RecapBaseModel.discriminator("Education", recapEducationSchema);

export const RecapEducationModel = mongoose.model<RecapEducationModel>("Education");

// Accomplishments

export interface RecapAccomplishments {
  title: string;
  type: string;
}

const recapAccomplishmentsSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
    maxlength: MAX_GENERAL_LENGTH,
  },
  type: {
    type: String,
    required: true,
    maxlength: MAX_GENERAL_LENGTH,
  },
});

export interface RecapAccomplishmentsModel extends RecapBaseModel, RecapAccomplishments {}

const RecapAccomplishments = RecapBaseModel.discriminator("Accomplishments", recapAccomplishmentsSchema);

export const RecapAccomplishmentsModel = mongoose.model<RecapAccomplishmentsModel>("Accomplishments");

// Skills

export interface RecapSkills {
  title: string;
}

const recapSkillsSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapSkillsModel extends RecapBaseModel, RecapSkills {}

const RecapSkills = RecapBaseModel.discriminator("Skills", recapSkillsSchema);

export const RecapSkillsModel = mongoose.model<RecapSkillsModel>("Skills");

// Side Projects

export interface RecapSideProjects {
  title: string;
}

const recapSideProjectsSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapSideProjectsModel extends RecapBaseModel, RecapSideProjects {}

const RecapSideProjects = RecapBaseModel.discriminator("SideProjects", recapSideProjectsSchema);

export const RecapSideProjectsModel = mongoose.model<RecapSideProjectsModel>("SideProjects");

// Organizations

export interface RecapOrganizations {
  organizationName: string;
  location: string;
  positions: string;
}

const recapOrganizationsSchema = new mongoose.Schema({
  organizationName: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  location: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  positions: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapOrganizationsModel extends RecapBaseModel, RecapOrganizations {}

const RecapOrganizations = RecapBaseModel.discriminator("Organizations", recapOrganizationsSchema);

export const RecapOrganizationsModel = mongoose.model<RecapOrganizationsModel>("Organizations");

// References

export interface RecapReferences {
  company: string;
  title: string;
  phoneNumber: string;
  email: string;
}

export const recapReferencesSchema = new mongoose.Schema({
  company: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  phoneNumber: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  email: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapReferencesModel extends RecapBaseModel, RecapReferences {}

const RecapReferences = RecapBaseModel.discriminator("References", recapReferencesSchema);

export const RecapReferencesModel = mongoose.model<RecapReferencesModel>("References");

// Other

export interface RecapOther {
  title: string;
}

const recapOtherSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapOtherModel extends RecapBaseModel, RecapOther {}

const RecapOther = RecapBaseModel.discriminator("Other", recapOtherSchema);

export const RecapOtherModel = mongoose.model<RecapOtherModel>("Other");
