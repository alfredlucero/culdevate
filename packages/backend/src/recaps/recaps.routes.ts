import { Router } from "express";
import RecapsController from "./recaps.controller";
import authMiddleware from "../middleware/auth.middleware";
import validationMiddleware from "../middleware/validation.middleware";
import { RecapSchema } from "./recaps.validation";

const RecapsRoutes = {
  path: "/recaps",
  router: Router(),

  initializeRoutes() {
    // POST /recaps
    this.router.post("/", authMiddleware, validationMiddleware(RecapSchema, "body"), RecapsController.createRecap);

    // PATCH /recaps/:recapId
    this.router.patch(
      "/:recapId",
      authMiddleware,
      validationMiddleware(RecapSchema, "body"),
      RecapsController.updateRecap,
    );

    return this.router;
  },
};

export default RecapsRoutes;
