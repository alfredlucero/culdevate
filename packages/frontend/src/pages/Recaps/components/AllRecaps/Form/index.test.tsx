import React from "react";
import { render } from "@testing-library/react";
import AllRecapsForm, { AllRecapsFormProps } from "./index";
import { RecapSkills, RecapKind } from "../../../recaps.interface";
import { RecapFields } from "../../../recaps.schema";

const initialRecap: RecapSkills = {
  title: "Tagalog",
  kind: RecapKind.Skills,
  userId: "userId",
  _id: "skillsId",
  proficiency: "Intermediate",
  bulletPoints: [],
};
const defaultProps: AllRecapsFormProps = {
  initialRecap,
  onSaveSuccess: () => {},
  isShowing: true,
  onHide: () => {},
  selectedKind: RecapKind.Skills,
};

describe("<AllRecapsForm />", () => {
  test("should show empty form inputs and disabled state given no initial recap", () => {
    const { container } = render(
      <AllRecapsForm {...defaultProps} initialRecap={null} testId="allRecapsFormTestId" className="extra-class-name" />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render pre-populated form inputs given initial recap", () => {
    const { container } = render(
      <AllRecapsForm
        {...defaultProps}
        initialRecap={initialRecap}
        testId="allRecapsFormTestId"
        className="extra-class-name"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should show work experience form given work experience is selected", () => {
    const { getByLabelText } = render(
      <AllRecapsForm {...defaultProps} selectedKind={RecapKind.WorkExperience} initialRecap={null} />,
    );

    expect(getByLabelText(RecapFields.workTitle)).toBeVisible();
  });

  test("should show education form given education is selected", () => {
    const { getByLabelText } = render(
      <AllRecapsForm {...defaultProps} selectedKind={RecapKind.Education} initialRecap={null} />,
    );

    expect(getByLabelText(RecapFields.educationSchool)).toBeVisible();
  });

  test("should show accomplishments form given accomplishments is selected", () => {
    const { getByLabelText } = render(
      <AllRecapsForm {...defaultProps} selectedKind={RecapKind.Accomplishments} initialRecap={null} />,
    );

    expect(getByLabelText(RecapFields.accomplishmentsTitle)).toBeVisible();
  });

  test("should show organizations form given organizations is selected", () => {
    const { getByLabelText } = render(
      <AllRecapsForm {...defaultProps} selectedKind={RecapKind.Organizations} initialRecap={null} />,
    );

    expect(getByLabelText(RecapFields.organizationsName)).toBeVisible();
  });

  test("should show skills form given skills is selected", () => {
    const { getByLabelText } = render(
      <AllRecapsForm {...defaultProps} selectedKind={RecapKind.Skills} initialRecap={null} />,
    );

    expect(getByLabelText(RecapFields.skillsTitle)).toBeVisible();
  });

  test("should show side projects form given side projects is selected", () => {
    const { getByLabelText } = render(
      <AllRecapsForm {...defaultProps} selectedKind={RecapKind.SideProjects} initialRecap={null} />,
    );

    expect(getByLabelText(RecapFields.sideProjectsTitle)).toBeVisible();
  });

  test("should show publications form given publications is selected", () => {
    const { getByLabelText } = render(
      <AllRecapsForm {...defaultProps} selectedKind={RecapKind.Publications} initialRecap={null} />,
    );

    expect(getByLabelText(RecapFields.publicationsTitle)).toBeVisible();
  });

  test("should show references form given references is selected", () => {
    const { getByLabelText } = render(
      <AllRecapsForm {...defaultProps} selectedKind={RecapKind.References} initialRecap={null} />,
    );

    expect(getByLabelText(RecapFields.referencesTitle)).toBeVisible();
  });

  test("should show other form given other is selected", () => {
    const { getByLabelText } = render(
      <AllRecapsForm {...defaultProps} selectedKind={RecapKind.Other} initialRecap={null} />,
    );

    expect(getByLabelText(RecapFields.otherTitle)).toBeVisible();
  });
});
