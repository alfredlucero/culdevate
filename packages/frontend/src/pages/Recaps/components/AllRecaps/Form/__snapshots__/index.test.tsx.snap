// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllRecapsForm /> should render pre-populated form inputs given initial recap 1`] = `
<div>
  <div
    class="mt-4 extra-class-name"
    data-testid="allRecapsFormTestId"
  >
    <form>
      <div
        class="flex w-full mb-4"
      >
        <div
          class="w-1/2 mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="title"
          >
            Title
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="title"
            name="title"
            placeholder="What skill did you learn?"
            type="text"
            value="Tagalog"
          />
        </div>
        <div
          class="w-1/2"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="proficiency"
          >
            Proficiency
             
            
          </label>
          <div
            class="inline-block relative"
          >
            <select
              aria-required="true"
              class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline"
              data-testid=""
              id="proficiency"
              name="proficiency"
              required=""
            >
              
              <option
                value="Novice"
              >
                Novice
              </option>
              <option
                value="Intermediate"
              >
                Intermediate
              </option>
              <option
                value="Advanced"
              >
                Advanced
              </option>
              <option
                value="Expert"
              >
                Expert
              </option>
            </select>
            <div
              class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
            >
              <svg
                class="fill-current h-4 w-4"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div>
        <span
          class="block text-gray-700 text-sm font-bold mb-2"
        >
          Bullet Points
        </span>
        <div
          class="mb-6"
        >
          <div
            class="bg-gray-300 pt-6 px-4 pb-4 bullet-point-list-droppable"
            data-rbd-droppable-context-id="1"
            data-rbd-droppable-id="bulletPointInputList"
          >
            <div
              aria-labelledby="rbd-lift-instruction-1"
              class="bg-white hover:bg-blue-100 rounded"
              data-rbd-drag-handle-context-id="1"
              data-rbd-drag-handle-draggable-id="skillsBulletPoint-0"
              data-rbd-draggable-context-id="1"
              data-rbd-draggable-id="skillsBulletPoint-0"
              draggable="false"
              tabindex="0"
            >
              <div
                class="flex items-start px-4 py-6 rounded shadow-md cursor-pointer mb-2"
              >
                <div
                  class="font-sans text-2xl mr-4 text-teal-300"
                  id=""
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-square fa-w-14 "
                    data-icon="square"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="mr-4 flex-1"
                >
                  
                  <input
                    aria-required="true"
                    class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
                    id="skillsBulletPoint-0"
                    name="skillsBulletPoint-0"
                    placeholder="Provide your highlights and results..."
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="font-sans text-3xl text-red-500 hover:text-red-700 cursor-pointer"
                  id="skillsBulletPoint-0"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-trash-alt fa-w-14 "
                    data-icon="trash-alt"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="flex justify-center"
            >
              <button
                class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mt-4 flex items-center"
                type="button"
              >
                <div
                  class="flex items-center"
                >
                  <div
                    class="font-sans text-2xl mr-2 add-bullet-point-plus-icon"
                    id=""
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-plus fa-w-14 "
                      data-icon="plus"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  Add Bullet Point
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-end"
      >
        <button
          class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mr-2"
          type="button"
        >
          <div
            class="flex items-center"
          >
            Cancel
          </div>
        </button>
        <button
          class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-teal-500 text-white hover:bg-teal-700 "
          data-testid="skillsSaveButton"
          type="submit"
        >
          <div
            class="flex items-center"
          >
            Save Recap
          </div>
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<AllRecapsForm /> should show empty form inputs and disabled state given no initial recap 1`] = `
<div>
  <div
    class="mt-4 extra-class-name"
    data-testid="allRecapsFormTestId"
  >
    <form>
      <div
        class="flex w-full mb-4"
      >
        <div
          class="w-1/2 mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="title"
          >
            Title
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="title"
            name="title"
            placeholder="What skill did you learn?"
            type="text"
            value=""
          />
        </div>
        <div
          class="w-1/2"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="proficiency"
          >
            Proficiency
             
            
          </label>
          <div
            class="inline-block relative"
          >
            <select
              aria-required="true"
              class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline"
              data-testid=""
              id="proficiency"
              name="proficiency"
              required=""
            >
              
              <option
                value="Novice"
              >
                Novice
              </option>
              <option
                value="Intermediate"
              >
                Intermediate
              </option>
              <option
                value="Advanced"
              >
                Advanced
              </option>
              <option
                value="Expert"
              >
                Expert
              </option>
            </select>
            <div
              class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
            >
              <svg
                class="fill-current h-4 w-4"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div>
        <span
          class="block text-gray-700 text-sm font-bold mb-2"
        >
          Bullet Points
        </span>
        <div
          class="mb-6"
        >
          <div
            class="bg-gray-300 pt-6 px-4 pb-4 bullet-point-list-droppable"
            data-rbd-droppable-context-id="0"
            data-rbd-droppable-id="bulletPointInputList"
          >
            <div
              aria-labelledby="rbd-lift-instruction-0"
              class="bg-white hover:bg-blue-100 rounded"
              data-rbd-drag-handle-context-id="0"
              data-rbd-drag-handle-draggable-id="skillsBulletPoint-0"
              data-rbd-draggable-context-id="0"
              data-rbd-draggable-id="skillsBulletPoint-0"
              draggable="false"
              tabindex="0"
            >
              <div
                class="flex items-start px-4 py-6 rounded shadow-md cursor-pointer mb-2"
              >
                <div
                  class="font-sans text-2xl mr-4 text-teal-300"
                  id=""
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-square fa-w-14 "
                    data-icon="square"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="mr-4 flex-1"
                >
                  
                  <input
                    aria-required="true"
                    class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
                    id="skillsBulletPoint-0"
                    name="skillsBulletPoint-0"
                    placeholder="Provide your highlights and results..."
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="font-sans text-3xl text-red-500 hover:text-red-700 cursor-pointer"
                  id="skillsBulletPoint-0"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-trash-alt fa-w-14 "
                    data-icon="trash-alt"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="flex justify-center"
            >
              <button
                class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mt-4 flex items-center"
                type="button"
              >
                <div
                  class="flex items-center"
                >
                  <div
                    class="font-sans text-2xl mr-2 add-bullet-point-plus-icon"
                    id=""
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-plus fa-w-14 "
                      data-icon="plus"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  Add Bullet Point
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-end"
      >
        <button
          class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mr-2"
          type="button"
        >
          <div
            class="flex items-center"
          >
            Cancel
          </div>
        </button>
        <button
          class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-teal-500 text-white hover:bg-teal-700 opacity-50 cursor-not-allowed "
          data-testid="skillsSaveButton"
          disabled=""
          type="submit"
        >
          <div
            class="flex items-center"
          >
            Save Recap
          </div>
        </button>
      </div>
    </form>
  </div>
</div>
`;
